src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -109,208 +95,60 @@ class MulticopterPositionControl : public control::SuperBlock, public ModulePara
-
-	math::LowPassFilter2p _filter_manual_pitch;
-	math::LowPassFilter2p _filter_manual_roll;
-
-	enum manual_stick_input {
-		brake,
-		direction_change,
-		acceleration,
-		deceleration
-	};
-
-	manual_stick_input _user_intention_xy; /**< defines what the user intends to do derived from the stick input */
-	manual_stick_input
-	_user_intention_z; /**< defines what the user intends to do derived from the stick input in z direciton */
-
-	matrix::Vector3f _pos_p;
-	matrix::Vector3f _vel_p;
-	matrix::Vector3f _vel_i;
-	matrix::Vector3f _vel_d;
-	float _tilt_max_air; /**< maximum tilt angle [rad] */
-	float _tilt_max_land; /**< maximum tilt angle during landing [rad] */
-	float _man_tilt_max;
-	float _man_yaw_max;
-	float _global_yaw_max;
-
-	struct map_projection_reference_s _ref_pos;
-	float _ref_alt;
-	bool _ref_alt_is_global; /** true when the reference altitude is defined in a global reference frame */
-	hrt_abstime _ref_timestamp;
-	hrt_abstime _last_warn;
-
-	matrix::Vector3f _thrust_int;
-	matrix::Vector3f _pos;
-	matrix::Vector3f _pos_sp;
-	matrix::Vector3f _vel;
-	matrix::Vector3f _vel_sp;
-	matrix::Vector3f _vel_prev;			/**< velocity on previous step */
-	matrix::Vector3f _vel_sp_prev;
-	matrix::Vector3f _vel_err_d;		/**< derivative of current velocity */
-	matrix::Vector3f _curr_pos_sp;  /**< current setpoint of the triplets */
-	matrix::Vector3f _prev_pos_sp; /**< previous setpoint of the triples */
-	matrix::Vector2f _stick_input_xy_prev; /**< for manual controlled mode to detect direction change */
-
-	matrix::Dcmf _R;			/**< rotation matrix from attitude quaternions */
-	float _yaw;				/**< yaw angle (euler) */
-	float _yaw_takeoff;	/**< home yaw angle present when vehicle was taking off (euler) */
-	float _man_yaw_offset; /**< current yaw offset in manual mode */
-
-	float _vel_max_xy;  /**< equal to vel_max except in auto mode when close to target */
-	bool _vel_sp_significant; /** true when the velocity setpoint is over 50% of the _vel_max_xy limit */
-	float _acceleration_state_dependent_xy; /**< acceleration limit applied in manual mode */
-	float _acceleration_state_dependent_z; /**< acceleration limit applied in manual mode in z */
-	float _manual_jerk_limit_xy; /**< jerk limit in manual mode dependent on stick input */
-	float _manual_jerk_limit_z; /**< jerk limit in manual mode in z */
-	float _z_derivative; /**< velocity in z that agrees with position rate */
-
-	float _takeoff_vel_limit; /**< velocity limit value which gets ramped up */
-
-	float _min_hagl_limit; /**< minimum continuous height above ground (m) */
-
-	// counters for reset events on position and velocity states
-	// they are used to identify a reset event
-	uint8_t _z_reset_counter;
-	uint8_t _xy_reset_counter;
-	uint8_t _heading_reset_counter;
-
-	matrix::Dcmf _R_setpoint;
+	hrt_abstime _last_warn = 0; /**< timer when the last warn message was sent out */
+	static constexpr uint64_t IDLE_BOFORE_TAKEOFF_TIME_US =
```

Comment:
Why should we wait for a hardcoded 2.5 seconds until the position controller realizes that the vehicle was armed?